execution_engine/src/core/engine_state/step.rs

```diff
@@ -0,0 +1,89 @@
+use std::vec::Vec;
+
+use casper_types::{bytesrepr, Key, ProtocolVersion, PublicKey, U512};
+
+use crate::shared::{newtypes::Blake2bHash, TypeMismatch};
+use casper_types::bytesrepr::FromBytes;
+use core::fmt;
+use std::fmt::Display;
+use uint::static_assertions::_core::fmt::Formatter;
+
+#[derive(Debug)]
+pub struct SlashItem {
+    pub validator_id: Vec<u8>,
+}
+
+impl SlashItem {
+    pub fn new(validator_id: Vec<u8>) -> Self {
+        Self { validator_id }
+    }
+}
+
+#[derive(Debug)]
+pub struct RewardItem {
+    pub validator_id: Vec<u8>,
```

Comment:
Usually we were keeping objects in similar objects rather than bytes: this could be `validator_id: casper_types::PublicKey` and then conversion could be done from within protobuf mapping code.